Result/Result.swift

```diff
@@ -179,6 +179,15 @@ public func ?? <T, Error> (left: Result<T, Error>, @autoclosure right: () -> Res
 	return left.recoverWith(right())
 }
 
+// MARK: - Derive result from failable closure
+
+public func materialize<T>(f: () throws -> T) -> Result<T, ErrorType> {
```

Comment:
OK, that's still acceptable for me. Thank you for your explanation.

The next issue is casting error to `E`. Because it's failable, we should instantiate `E` somehow if casting fails. Is there any way to instantiate `E` in this case?

```swift
public func materialize<T, E: ErrorType>(f: () throws -> T) -> Result<T, E> {
    do {
        return .success(try f())
    } catch let error as E {
		return .failure(error)
	} catch {
		// what should we return here?
	}
}
```